### PR TITLE
docs: document cluster config V1

### PR DESCRIPTION
**Pull request by corverroos**
Documents and spec V1 cluster config which formalises ideas introduced in dkg doc. 

DKG doc needs to be updated to aligned with this.

category: docs 
ticket: none

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -2,6 +2,66 @@
...
+
+The `cluster.lock` has the following schema:
+```json
+{
+  "cluster_config": {...},  // Cluster json config, identical schema to above,
+  "distributed_validators": [                               // Length equaled to num_validators.
+    {
+      "distributed_public_key":  "0x123..abfc",             // DV root pubkey
+      "threshold_verifiers": [ "oA8Z...2XyT", "g1q...icu"], // length of threshold
+      "fee_recipient": "0x123..abfc"                        // Empty by default, can be edited manually
```
Comment (OisinKyne) [inline docs/configuration.md:51]: I would reassure people with:

`// Defaults to withdrawal address if not set, can be edited manually`

**Review by OisinKyne — APPROVED** [review]

**Comment by OisinKyne** [comment]
LGTM

File: docs/configuration.md [hunk 2]
```diff
@@ -2,6 +2,66 @@
...
+- `cluster.lock` which includes `cluster.json` and includes distributed validator keys.
+
+The `charon create dkg` command is used to create `cluster.json` file which is used as input to `charon dkg`.
+
+The `charon create cluster` command combines both steps into one and just outputs the final `cluster.lock` without a DKG step.
+
+The schema of the `cluster.json` is defined as:
+```json
+{
+  "version": "1.0.0",                   // Schema version
```
[the remark this answers is not in the document]
Comment (corverroos) [reply docs/configuration.md:20]: sure will add v prefix 👍

File: docs/configuration.md (hunk 1 above)
Comment (corverroos) [reply docs/configuration.md:51]:
```suggestion
      "fee_recipient": "0x123..abfc"                        // Defaults to withdrawal address if not set, can be edited manually
```

File: docs/configuration.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (corverroos) [reply docs/configuration.md:20]:
```suggestion
  "version": "v1.0.0",                   // Schema version
```

**Pull request merged**